### PR TITLE
Feature/enhance data migration

### PR DESCRIPTION
**Pull request by Marianasilvago**
Changes:

Add validation to zebedee users migration script and upload files to S3

How to check?

Verify the code changes

Who can check?
Anyone.

### CODE REVIEW

**Review by AWitcherONS — DISMISSED** [review]
Just one spelling mistake given that you have found more of mine

File: dp-identity-data-migration/userdataextraction/user_extraction.go [hunk 1]
```diff
@@ -144,10 +170,54 @@ func process_zebedee_users(csvwriter *csv.Writer, userlist []zebedee.User) {
 		csvline.mfa_enabled = "FALSE"
 		csvline.phone_number_verified = "FALSE"
 		csvline.email_verified = "TRUE"
-		if err := csvwriter.Write(convert_to_slice(csvline)); err != nil {
-			fmt.Println("error writing record to csv:", err)
+
+		if isValidEmail := validateEmailId(validEmailDomains, user.Email); isValidEmail {
```
Comment (rafahop) [inline dp-identity-data-migration/userdataextraction/user_extraction.go:174]: Do we need a variable?

```suggestion
		if validateEmailId(validEmailDomains, user.Email) {
```

File: dp-identity-data-migration/userdataextraction/user_extraction.go [hunk 2]
```diff
@@ -144,10 +170,54 @@ func process_zebedee_users(csvwriter *csv.Writer, userlist []zebedee.User) {
...
+			if err := validUsersWriter.Write(convert_to_slice(csvline)); err != nil {
+				fmt.Println("error writing record to csv:", err)
+			}else {
+				validUsersCount += 1
+			}
+		} else {
+			if err := invalidUsersWriter.Write(convert_to_slice(csvline)); err != nil {
```
Comment (rafahop) [inline dp-identity-data-migration/userdataextraction/user_extraction.go:181]: `convert_to_slice(csvline)` is run no matter what (in both sides of the if) so maybe consider moving it up to line 173 and use a variable?

File: dp-identity-data-migration/userdataextraction/user_extraction.go [hunk 3]
```diff
@@ -144,10 +170,54 @@ func process_zebedee_users(csvwriter *csv.Writer, userlist []zebedee.User) {
...
+			}else {
+				invalidUsersCount += 1
+			}
 		}
 	}
+	return validUsersCount, invalidUsersCount
+}
+
+func validateEmailId(validEmailDomains []string, emailID string) bool {
+	domainName := strings.Split(emailID, "@")[1]
```
Comment (rafahop) [inline dp-identity-data-migration/userdataextraction/user_extraction.go:192]: What if the `emailID` did not contain the `@` symbol?

File: dp-identity-data-migration/userdataextraction/user_extraction.go [hunk 4]
```diff
@@ -144,10 +170,54 @@ func process_zebedee_users(csvwriter *csv.Writer, userlist []zebedee.User) {
...
+	for _, domain := range validEmailDomains {
+		if strings.Contains(domain, domainName) {
```
Comment (rafahop) [inline dp-identity-data-migration/userdataextraction/user_extraction.go:194]: I wonder whether we should use exact matches?

**Review by rafahop — CHANGES_REQUESTED** [review]
Please run `go fmt` for both go files

**Review by rafahop — APPROVED** [review]
Looks good to me

**Pull request merged**